src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json

```diff
@@ -126,5 +126,53 @@
     "destKey": "user.user_agent",
     "sourceKeys": ["properties.context.user.userAgent", "context.userAgent"],
     "required": false
+  },
+  {
+    "destKey": "user.first_name",
+    "sourceKeys": "firstName",
+    "metadata": {
+      "type": ["trim", "toLower", "hashToSha256"]
+    },
+    "sourceFromGenericMap": true
+  },
+  {
+    "destKey": "user.last_name",
+    "sourceKeys": "lastName",
+    "metadata": {
+      "type": ["trim", "toLower", "hashToSha256"]
+    },
+    "sourceFromGenericMap": true
+  },
+  {
+    "destKey": "user.city",
+    "sourceKeys": ["context.traits.city", "context.traits.address.city", "properties.city"]
+  },
+  {
+    "destKey": "user.country",
+    "sourceKeys": ["context.traits.country", "context.traits.address.country", "properties.country"]
+  },
+  {
+    "destKey": "user.state",
+    "sourceKeys": ["context.traits.address.state", "context.traits.state", "properties.state"]
+  },
+  {
+    "destKey": "user.zip_code",
+    "sourceKeys": [
+      "context.traits.zip",
+      "context.traits.zip_code",
+      "context.traits.postalcode",
+      "context.traits.postalCode",
+      "context.traits.address.zip",
+      "context.traits.address.zip_code",
+      "context.traits.address.postalcode",
+      "context.traits.address.postalCode",
+      "properties.zip",
+      "properties.postalcode",
+      "properties.postalCode",
+      "properties.zip_code"
+    ],
```

Comment:
We already have generic mapping right ?! Would that not suffice ?

Reply:
src/v0/util/data/GenericFieldMapping.json

Reply:
Check if we can use genericMapping for city, state, country fields

Reply:
changed for zipcode and others won't suffice